homeassistant/components/getsequence/coordinator.py

```diff
@@ -0,0 +1,168 @@
+"""DataUpdateCoordinator for Sequence integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import aiohttp
+from GetSequenceIoApiClient import SequenceApiClient, SequenceApiError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import (
+    TimestampDataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    CONF_INVESTMENT_ACCOUNTS,
+    CONF_LIABILITY_ACCOUNTS,
+    CONF_LIABILITY_CONFIGURED,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SequenceDataUpdateCoordinator(TimestampDataUpdateCoordinator):
+    """Class to manage fetching data from the Sequence API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        session: aiohttp.ClientSession,
+        update_interval: timedelta,
+    ) -> None:
+        """Initialize the coordinator."""
+        self.api = SequenceApiClient(session, entry.data[CONF_ACCESS_TOKEN])
+        self.entry = entry
```

Comment:
```suggestion
    """Class to manage fetching data from the Sequence API."""
    
    config_entry: SequenceConfigEntry

    def __init__(
        self,
        hass: HomeAssistant,
        entry: SequenceConfigEntry,
        session: aiohttp.ClientSession,
        update_interval: timedelta,
    ) -> None:
        """Initialize the coordinator."""
        self.api = SequenceApiClient(session, entry.data[CONF_ACCESS_TOKEN])
```
the super constructor sets that